user-operator/src/main/java/io/strimzi/operator/user/UserOperatorConfig.java

```diff
@@ -4,554 +4,407 @@
+import static io.strimzi.operator.common.operator.resource.ConfigParameterParser.BOOLEAN;
 
 /**
  * Cluster Operator configuration
  */
 public class UserOperatorConfig {
-    // Environment variable names
-    static final String STRIMZI_NAMESPACE = "STRIMZI_NAMESPACE";
-    static final String STRIMZI_FULL_RECONCILIATION_INTERVAL_MS = "STRIMZI_FULL_RECONCILIATION_INTERVAL_MS";
-    static final String STRIMZI_LABELS = "STRIMZI_LABELS";
-    static final String STRIMZI_CA_CERT_SECRET_NAME = "STRIMZI_CA_CERT_NAME";
-    static final String STRIMZI_CA_KEY_SECRET_NAME = "STRIMZI_CA_KEY_NAME";
-    static final String STRIMZI_CLUSTER_CA_CERT_SECRET_NAME = "STRIMZI_CLUSTER_CA_CERT_SECRET_NAME";
-    static final String STRIMZI_EO_KEY_SECRET_NAME = "STRIMZI_EO_KEY_SECRET_NAME";
-    static final String STRIMZI_CA_NAMESPACE = "STRIMZI_CA_NAMESPACE";
-    static final String STRIMZI_KAFKA_BOOTSTRAP_SERVERS = "STRIMZI_KAFKA_BOOTSTRAP_SERVERS";
-    static final String STRIMZI_CLIENTS_CA_VALIDITY = "STRIMZI_CA_VALIDITY";
-    static final String STRIMZI_CLIENTS_CA_RENEWAL = "STRIMZI_CA_RENEWAL";
-    static final String STRIMZI_SECRET_PREFIX = "STRIMZI_SECRET_PREFIX";
-    static final String STRIMZI_ACLS_ADMIN_API_SUPPORTED = "STRIMZI_ACLS_ADMIN_API_SUPPORTED";
-    static final String STRIMZI_KRAFT_ENABLED = "STRIMZI_KRAFT_ENABLED";
-    static final String STRIMZI_SCRAM_SHA_PASSWORD_LENGTH = "STRIMZI_SCRAM_SHA_PASSWORD_LENGTH";
-    static final String STRIMZI_MAINTENANCE_TIME_WINDOWS = "STRIMZI_MAINTENANCE_TIME_WINDOWS";
-    static final String STRIMZI_KAFKA_ADMIN_CLIENT_CONFIGURATION = "STRIMZI_KAFKA_ADMIN_CLIENT_CONFIGURATION";
-    static final String STRIMZI_OPERATION_TIMEOUT_MS = "STRIMZI_OPERATION_TIMEOUT_MS";
-    static final String STRIMZI_WORK_QUEUE_SIZE = "STRIMZI_WORK_QUEUE_SIZE";
-    static final String STRIMZI_CONTROLLER_THREAD_POOL_SIZE = "STRIMZI_CONTROLLER_THREAD_POOL_SIZE";
-    static final String STRIMZI_CACHE_REFRESH_INTERVAL_MS = "STRIMZI_CACHE_REFRESH_INTERVAL_MS";
-    static final String STRIMZI_BATCH_QUEUE_SIZE = "STRIMZI_BATCH_QUEUE_SIZE";
-    static final String STRIMZI_BATCH_MAXIMUM_BLOCK_SIZE = "STRIMZI_BATCH_MAXIMUM_BLOCK_SIZE";
-    static final String STRIMZI_BATCH_MAXIMUM_BLOCK_TIME_MS = "STRIMZI_BATCH_MAXIMUM_BLOCK_TIME_MS";
-    static final String STRIMZI_USER_OPERATIONS_THREAD_POOL_SIZE = "STRIMZI_USER_OPERATIONS_THREAD_POOL_SIZE";
-
-    // Default values
-    static final long DEFAULT_FULL_RECONCILIATION_INTERVAL_MS = 120_000;
-    static final String DEFAULT_KAFKA_BOOTSTRAP_SERVERS = "localhost:9091";
+
+    private static final Map<String, ConfigParameter<?>> CONFIG_VALUES = new HashMap<>();
+
+    /**
+     * Namespace in which the operator will run and create resources
+     */
+    public static final ConfigParameter<String> NAMESPACE = new ConfigParameter<>("STRIMZI_NAMESPACE", NON_EMPTY_STRING, "", true, CONFIG_VALUES);
```

Comment:
I would really like to see two static factory methods on `ConfigParameter`:

```java
public static ConfigParameter<T> required(String key, ConfigParameterParser<T> type, Map<String, ConfigParameter<?>> map) { 
/* just calls the constructor */ 
}
ConfigParameter<T> optional(String key, ConfigParameterParser type, String defaultValue, Map<String, public static ConfigParameter<?>> map) {
 /* just calls the constructor */
}
````

That will make all these declarations much clearer, e.g.

```java
public static final ConfigParameter<String> NAMESPACE =  ConfigParameter.required("STRIMZI_NAMESPACE", NON_EMPTY_STRING, CONFIG_VALUES);
```

 and also meansthe reader doesn't have to do a double-take when they see a default of `""` for something with `NON_EMPTY_STRING` type. ("Oh, it's OK, the default is never used because it's a required paramete")